asset-bundle-converter/Assets/AssetBundleConverter/VisualTests.cs

```diff
@@ -349,9 +349,13 @@ public static GameObject[] LoadAndInstantiateAllAssetBundles(ClientSettings Clie
                             material.shader = Shader.Find("Shader Graphs/glTF-pbrMetallicRoughness");
                         }
 
-
-                        SRPBatchingHelper.OptimizeMaterial(material);
-
+                        #if UNITY_EDITOR
+                            if (UnityEditor.EditorUserBuildSettings.selectedBuildTargetGroup == UnityEditor.BuildTargetGroup.WebGL)
+                                SRPBatchingHelper.OptimizeMaterial(material);

```

Comment:
I would inject a boolean indicating the platform when instanciating `AssetBundleConverterMaterialGenerator`. Making the `#if` branching in two different places may be harder to maintain.

Reply:
How about I put it in the optimizematerial function itself?